src/Http/Driver/CurlDriver.php

```diff
@@ -60,7 +59,7 @@ public function execute(RequestInterface $request)
             $headers[] = $header.': '.implode(', ', $values);
         }
         curl_setopt(self::$ch, CURLOPT_HTTPHEADER, $headers);
-        curl_setopt(self::$ch, CURLOPT_POSTFIELDS, $request->getBody()->getContents());
+        curl_setopt(self::$ch, CURLOPT_POSTFIELDS, (string) $request->getBody());
```

Comment:
A request may contain a large stream: if you need to stream from the beginning of it, then you need to rely on a string cast, or else it will read from the current position in the stream (usually the end, causing an empty string to be read)